src/roman_datamodels/stnode/_tagged.py

```diff
@@ -78,12 +85,82 @@ def create_minimal(cls, defaults=None, builder=None, *, tag: str | None = None):
         return new
 
     @classmethod
-    def create_fake_data(cls, defaults=None, shape=None, builder=None, *, tag: str | None = None):
-        return cls.create_minimal(defaults, builder or FakeDataBuilder(shape), tag=tag)
+    def create_minimal(cls, defaults: Mapping[str, Any] | None = None, *, tag: str | None = None) -> Self:
+        """
+        Create a minimal instance of this class, only things with the attributes
+        which have a default value that can be determined.
+
+        Parameters
+        ----------
+        defaults : Mapping[str, Any] | None
+            A mapping of default values to use when creating the instance
+        tag : str | None
+            The tag to use when creating the instance. If None, the default tag for the class will be used.
+
+        Returns
+        -------
+        Self
+            An instance of this class
+        """
+        return cls._create_minimal(defaults, tag=tag)
+
+    @classmethod
+    def _create_fake_data(
+        cls,
+        defaults: Mapping[str, Any] | None = None,
+        shape: tuple[int, ...] | None = None,
+        builder: Builder | None = None,
+        *,
+        tag: str | None = None,
+    ) -> Self:
+        return cls._create_minimal(defaults, builder or FakeDataBuilder(shape), tag=tag)
+
+    @classmethod
+    def create_fake_data(
+        cls, defaults: Mapping[str, Any] | None = None, shape: tuple[int, ...] | None = None, *, tag: str | None = None
```

Comment:
```suggestion
        cls, *, defaults: Mapping[str, Any] | None = None, shape: tuple[int, ...] | None = None, tag: str | None = None
```

Ideally, I would prefer to make these keyword only inputs as things like this function really don't have a good defined of inputs you would want to pass. It seems just as prudent to pass `shape` only as it is `defaults` only.  In general, I am in favor of making function parameters with default values keyword only whenever possible.